packages/palette/src/elements/MultiSelect/MultiSelect.tsx

```diff
@@ -0,0 +1,172 @@
+import { themeGet } from "@styled-system/theme-get"
+import React, { useEffect, useState } from "react"
+import styled from "styled-components"
+import { DROP_SHADOW } from "../../helpers"
+import { useClickOutside, usePosition } from "../../utils"
+import { useUpdateEffect } from "../../utils/useUpdateEffect"
+import { Box, BoxProps } from "../Box"
+import { Checkbox } from "../Checkbox"
+import { Clickable } from "../Clickable"
+import { caretMixin, Option } from "../Select"
+import { Sup } from "../Sup"
+import { Text } from "../Text"
+
+export interface MultiSelectProps extends BoxProps {
+  title: React.ReactNode
+  options: Option[]
+  onSelect?: (selection: Option[]) => void
+}
+
+export const MultiSelect: React.FC<MultiSelectProps> = ({
+  title,
+  options,
+  onSelect,
+  ...rest
+}) => {
+  const [visible, setVisible] = useState(false)
+  const [selection, setSelection] = useState<Option[]>([])
+  const [width, setWidth] = useState(0)
+
+  // Yields focus back and forth between popover and anchor
+  useUpdateEffect(() => {
+    if (visible && tooltipRef.current) {
+      tooltipRef.current.focus()
+      return
+    }
+
+    if (!anchorRef.current) return
+    anchorRef.current.focus()
+  }, [visible])
+
+  const onVisible = () => setVisible(true)
+  const onHide = () => setVisible(false)
+
+  useEffect(() => {
+    const handleKeydown = (event: KeyboardEvent) => {
+      if (event.key === "Escape") {
+        onHide()
+      }
+    }
+
+    document.addEventListener("keydown", handleKeydown)
+    return () => {
+      document.removeEventListener("keydown", handleKeydown)
+    }
+  }, [])
+
+  useEffect(() => {
+    if (!anchorRef.current) return
+
+    const handleResize = () => {
+      setWidth(anchorRef.current.offsetWidth)
+    }
+
+    handleResize()
+
+    window.addEventListener("resize", handleResize)
+    return () => {
+      window.removeEventListener("resize", handleResize)
+    }
+  }, [])
+
+  const { anchorRef, tooltipRef } = usePosition({
+    position: "bottom",
+    offset: 10,
+    active: visible,
+  })
+
+  useClickOutside({
+    ref: tooltipRef,
+    onClickOutside: onHide,
+    when: visible,
+    type: "click",
+  })
+
+  const handleSelect = (option: Option) => {
+    return (selected: boolean) => {
+      const prevSelection = selection
+      const nextSelection = selected
+        ? [...prevSelection, option]
+        : prevSelection.filter(({ value }) => option.value !== value)
+
+      setSelection(nextSelection)
+      onSelect?.(nextSelection)
+    }
+  }
+
+  return (
+    <>
+      <Container ref={anchorRef as any} onClick={onVisible} {...rest}>
+        <Text variant="sm" lineHeight={1}>
```

Comment:
Can we change the color here to match the `Input` component and the Figma design? 

```suggestion
        <Text variant="sm" lineHeight={1} color"black60">
```

Reply:
I just tried to pass a component for the `title` and it breaks the layout if we don't set `display: inline;`. Maybe we can wrap `{title}` with an inline component to prevent the line break.

Reply:
We should just restrict it to string, probably. Why do you need to pass it a component? Can you give me your use-case?

Reply:
The other thing I'll have to do is rename this prop from title to label maybe? 

![](https://static.damonzucconi.com/_capture/YbzAmvFb.png) — it looks like we have to keep the title/subtitle consistent with other inputs.

Reply:
The other thing I'll have to do is rename this prop from title to label maybe? 

![](https://static.damonzucconi.com/_capture/YbzAmvFb.png) — it looks like we have to keep the title/subtitle consistent with other inputs.

Reply:
Maybe `name`? `label` kind of seems like what `title` _should_ be called; but I'm having to keep it consistent since historically that's been title

Reply:
I've renamed `title` to `name` then added `title` and `description` support to match the other select/inputs. I'm still not sure what the need for a custom component is for what's now `name` but let me know and we'll figure it out.